### DISCUSSION TITLE
Can the ID field on the single-entity generated resolvers be extended to be an object? (Multiple Primary Columns)

### DISCUSSION THREAD

**Discussion by MateoGuerreroE** [TriPSs/nestjs-query#338] [Q&A] [answered]
Hey! So I'm having an issue while trying to use the single-entity generated endpoint from Nestjs-Query. I understand the basic structure of that endpoint would be `entity (id: "id") {...} `. However since the entity I'm working has several @PrimaryColumn decorators, TypeORM will complain with: `Cannot use given entity id \"id\" because \"EntityRecord\" contains multiple primary columns, you must provide object in following form: {\"id1\":1,\"id2\":2,\"id3\":3} as an id.`

This error comes from TypeORM, but looks like (and as per my investigation) It's expecting an object there, which can be extended If the resolver is manually made but since these resolvers are generated, is there a way for me to tell the generated resolver to extend the ID to an object, so It can process that object? (Bwt I already tried with stringify JSON and does nto work either, typeorm It's expecting an actual object)

Thanks :)

**Comment by TriPSs** [comment]
Maybe checkout [IDField](https://tripss.github.io/nestjs-query/docs/graphql/dtos/#idfield). Not sure if it works with multiple fields, but you can always try

**Reply by MateoGuerreroE** [reply]
Tried but for some reason even defining custom sacalar for the IDField, It does not recognize It correctly when It's an object, will keep trying a bit more.

**Reply by MateoGuerreroE** [reply, marked as answer]
For solution of this ^ since we use views to connect to generated resolvers, I created a compound id using the entity ids `${id1}-${id2}-...` and used that as unique primary key.
